From 16420637a27fe44c954bf0d6e0ef751d6fd4ff9b Mon Sep 17 00:00:00 2001
From: Erwan Gouriou <erwan.gouriou@linaro.org>
Date: Thu, 23 Sep 2021 14:19:25 +0200
Subject: [PATCH 07/12] tfc/target: stm32: Fix missing DBG clock on stm32g0/l0

On stm32g0 and stm32l0 series, DGCMCU clock should be enabled
before any access.
This was missing, fix this.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>
---
 tcl/target/stm32g0x.cfg | 4 ++++
 tcl/target/stm32l0.cfg  | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/tcl/target/stm32g0x.cfg b/tcl/target/stm32g0x.cfg
index b6d9a22a2..cda21ee2b 100644
--- a/tcl/target/stm32g0x.cfg
+++ b/tcl/target/stm32g0x.cfg
@@ -65,6 +65,10 @@ proc stm32g0x_default_reset_start {} {
 }
 
 proc stm32g0x_default_examine_end {} {
+	# Enable DBG clock
+	# RCC_APB1ENR |= DBGEN
+	mmw 0x4002103C 0x08000000 0
+
 	# DBGMCU_CR |= DBG_STANDBY | DBG_STOP
 	mmw 0x40015804 0x00000006 0
 
diff --git a/tcl/target/stm32l0.cfg b/tcl/target/stm32l0.cfg
index b4bdb18a4..d5e68220e 100644
--- a/tcl/target/stm32l0.cfg
+++ b/tcl/target/stm32l0.cfg
@@ -84,6 +84,10 @@ $_TARGETNAME configure -event reset-start {
 }
 
 $_TARGETNAME configure -event examine-end {
+	# Enable DBG clock
+	# RCC_APB2ENR |= DBGMCUEN
+	mmw 0x40001024 0x00400000 0
+
 	# DBGMCU_CR |= DBG_STANDBY | DBG_STOP | DBG_SLEEP
 	mmw 0x40015804 0x00000007 0
 
-- 
2.40.1

